src/app/main/shared/services/spinner.service.ts

```diff
@@ -0,0 +1,44 @@
+import {Injectable} from '@angular/core';
+import {LoggerFactory} from '../../../shared/logger/logger.factory';
+import {spinnerDefaultConfig} from '../config/spinner.default.config';
+
+@Injectable()
+export class SpinnerService {
+
+  private timeout = spinnerDefaultConfig.timeout;
+
+  private tasksCounter = 0;
+  private takesTooMuchTime = false;
+
+  constructor() {
+  }
+
+  addTask(): void {
+    this.tasksCounter++;
+    setTimeout(() => {
+      if (this.hasTasks()) {
+        this.takesTooMuchTime = true;
+      }
+    }, this.timeout);
```

Comment:
spinnerDefaultConfig.timeout can we use instead of this.timeout

Reply:
off course we can. done